packages/mobile/src/components/core/UserGeneratedText.tsx

```diff
@@ -65,20 +86,46 @@ const Link = ({ children, url, ...other }: TextLinkProps & { url: string }) => {
       if (res.data) {
         if (instanceOfTrackResponse(res)) {
           setUnfurledContent(formatTrackName({ track: res.data }))
+          setTo({ screen: 'Track', params: { id: decodeHashId(res.data.id) } })
         } else if (instanceOfPlaylistResponse(res)) {
           setUnfurledContent(formatCollectionName({ collection: res.data[0] }))
+          setTo({
+            screen: 'Collection',
+            params: { id: decodeHashId(res.data[0].id) }
+          })
         } else if (instanceOfUserResponse(res)) {
           setUnfurledContent(formatUserName({ user: res.data }))
+          setTo({
+            screen: 'Profile',
+            params: { id: decodeHashId(res.data.id) }
+          })
         }
       }
     }
   }, [url, shouldUnfurl, unfurledContent, setUnfurledContent])
 
-  return (
-    <TextLink {...other} url={url}>
-      {unfurledContent ?? url}
-    </TextLink>
+  const handlePress = useCallback(
+    (e: GestureResponderEvent) => {
+      if (to) {
+        if ('push' in navigation) {
+          // @ts-ignore
```

Comment:
why the ignores?

Reply:
Basically the typing with react-navigation is very difficult and breaks down at a certain level, especially when dealing with ambiguous navigator types. I probably could figure out the correct type but the investment didn't seem worth it